cpu/native/Makefile.dep

```diff
@@ -18,7 +18,8 @@ ifeq (,$(filter stdio_%,$(USEMODULE)))
 endif
 
 ifneq (,$(filter periph_rtc,$(USEMODULE)))
-  USEMODULE += xtimer
+  USEMODULE += ztimer
+  USEMODULE += ztimer_msec
```

Comment:
Kconfig needs an update too

Reply:
The Kconfig change is not straightforward: it creates a circular dependency (periph_rtc -> ztimer -> periph_rtc).

Reply:
I pushed the change but it won't work. This is what I get locally:

<details>

```
$ TEST_KCONFIG=1 make BOARD=native -C tests/periph_rtc --no-print-directory 
=== [ATTENTION] Testing Kconfig dependency modelling ===
Traceback (most recent call last):
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 443, in <module>
    main()
  File "/work/riot/RIOT/dist/tools/kconfiglib/genconfig.py", line 390, in main
    kconf = RiotKconfig(args.kconfig_filename, warn_to_stderr=False)
  File "/work/riot/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 947, in __init__
    self._init(filename, warn, warn_to_stderr, encoding)
  File "/work/riot/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 1120, in _init
    check_dep_loop_sym(sym, False)
  File "/work/riot/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 6606, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/work/riot/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 6606, in _check_dep_loop_sym
    else _check_dep_loop_sym(dep, False)
  File "/work/riot/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 6610, in _check_dep_loop_sym
    return _found_dep_loop(loop, sym)
  File "/work/riot/RIOT/dist/tools/kconfiglib/kconfiglib.py", line 6714, in _found_dep_loop
    raise KconfigError(msg)
kconfiglib.KconfigError: 
Dependency loop
===============

MODULE_ZTIMER (defined at /work/riot/RIOT/sys/ztimer/Kconfig:8), with definition...

menuconfig MODULE_ZTIMER
	bool "ztimer - High level timer abstraction layer"
	select MODULE_ZTIMER_CORE
	select MODULE_ZTIMER_CONVERT_FRAC
	select MODULE_ZTIMER_CONVERT_SHIFT
	select MODULE_ZTIMER_EXTEND
	depends on TEST_KCONFIG

(select-related dependencies: (MODULE_ILI9341 && HAS_PERIPH_SPI && HAS_PERIPH_GPIO && TEST_KCONFIG) || (MODULE_SX127X && HAS_PERIPH_GPIO && HAS_PERIPH_GPIO_IRQ && HAS_PERIPH_SPI && TEST_KCONFIG) || (MODULE_PERIPH_RTC && CPU_ARCH_NATIVE && HAS_PERIPH_RTC && TEST_KCONFIG) || (MODULE_SGP30 && HAS_PERIPH_I2C && TEST_KCONFIG) || (MODULE_SM_PWM_01C && HAS_PERIPH_GPIO && HAS_PERIPH_GPIO_IRQ && TEST_KCONFIG) || (MODULE_ARDUINO && HAS_ARDUINO && HAS_PERIPH_GPIO && HAS_CPP && TEST_KCONFIG) || (PACKAGE_DRIVER_SX126X && TEST_KCONFIG && HAS_PERIPH_SPI) || (MODULE_LVGL_CONTRIB && PACKAGE_LVGL && TEST_KCONFIG) || (PACKAGE_MYNEWT-CORE && TEST_KCONFIG && HAS_PERIPH_GPIO && !HAS_ARCH_ESP))

...depends on MODULE_PERIPH_RTC (defined at /work/riot/RIOT/drivers/periph_common/Kconfig.rtc:8, /work/riot/RIOT/cpu/native/periph/Kconfig.rtc:8), with definition...

config MODULE_PERIPH_RTC
	bool "RTC peripheral driver"
	select MODULE_PERIPH_COMMON
	depends on HAS_PERIPH_RTC && TEST_KCONFIG

config MODULE_PERIPH_RTC
	bool
	select MODULE_ZTIMER if CPU_ARCH_NATIVE
	select MODULE_ZTIMER_MSEC if CPU_ARCH_NATIVE
	depends on HAS_PERIPH_RTC && TEST_KCONFIG

(select-related dependencies: MODULE_ZTIMER_PERIPH_RTC && HAS_PERIPH_RTC && MODULE_ZTIMER)

...depends on MODULE_ZTIMER_PERIPH_RTC (defined at /work/riot/RIOT/sys/ztimer/Kconfig:24), with definition...

config MODULE_ZTIMER_PERIPH_RTC
	bool "RTC peripheral" if ZTIMER_CUSTOM_BACKEND_CONFIGURATION
	select MODULE_PERIPH_RTC
	depends on HAS_PERIPH_RTC && MODULE_ZTIMER

(select-related dependencies: ZTIMER_SEC_BACKEND_RTC && HAS_PERIPH_RTC && <choice>)

...depends on ZTIMER_CUSTOM_BACKEND_CONFIGURATION (defined at /work/riot/RIOT/sys/ztimer/Kconfig:18), with definition...

config ZTIMER_CUSTOM_BACKEND_CONFIGURATION
	bool "Override default backend selection"
	depends on MODULE_ZTIMER

...depends again on MODULE_ZTIMER (defined at /work/riot/RIOT/sys/ztimer/Kconfig:8)
```

</details>

@leandrolanzieri @fjmolinas do you have any idea how I could fix this ?
Can we expect the same kind of issues with the eth driver of stm32/sam0 which relies on xtimer for now ?

Reply:
> @leandrolanzieri @fjmolinas do you have any idea how I could fix this ?

We will probably need to adapt the dependencies in Ztimer to avoid this loop. I'll take a look into it.

> Can we expect the same kind of issues with the eth driver of stm32/sam0 which relies on xtimer for now ?

I think those are not yet modelled in Kconfig.

Reply:
@aabadie can you give it a try with this:
```diff
diff --git a/Kconfig b/Kconfig
index 694c37d85f..523d80a34b 100644
--- a/Kconfig
+++ b/Kconfig
@@ -15,6 +15,9 @@ rsource "kconfigs/Kconfig.features"
 # Load error symbols
 rsource "kconfigs/Kconfig.errors"
 
+# Load Kconfig macros
+rsource "kconfigs/Kconfig.macros"
+
 # For now, get used modules as macros from this file (see kconfig.mk)
 osource "$(KCONFIG_GENERATED_DEPENDENCIES)"
 
diff --git a/cpu/native/periph/Kconfig.rtc b/cpu/native/periph/Kconfig.rtc
index d55a7de808..816e72e999 100644
--- a/cpu/native/periph/Kconfig.rtc
+++ b/cpu/native/periph/Kconfig.rtc
@@ -6,5 +6,5 @@
 #
 
 config MODULE_PERIPH_RTC
-    select MODULE_XTIMER if CPU_ARCH_NATIVE
-    depends on HAS_PERIPH_RTC
+    select MODULE_ZTIMER if CPU_ARCH_NATIVE
+    select MODULE_ZTIMER_MSEC if CPU_ARCH_NATIVE
diff --git a/kconfigs/Kconfig.macros b/kconfigs/Kconfig.macros
new file mode 100644
index 0000000000..bc6d8a33dc
--- /dev/null
+++ b/kconfigs/Kconfig.macros
@@ -0,0 +1,26 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+# Kconfig macro functions.
+#
+# These functions are evaluated at parsing time when loading Kconfig files.
+# For more information see https://www.kernel.org/doc/html/latest/kbuild/kconfig-macro-language.html
+#
+
+# Compares two string values
+#
+# parameters:
+#   1. first value to compare
+#   2. second value to compare
+#   3. value returned when 1 and 2 match
+#   4. value returned when 1 and 2 do not match
+#
+# usage:
+#   - $(if-eq,foo,foo,y) -> resolves to y
+#   - $(if-eq,foo,bar,,n) -> resolves to n
+#
+if-eq = $(shell, [[ $(1) = $(2) ]] && echo $(3) || echo $(4))
diff --git a/sys/ztimer/Kconfig b/sys/ztimer/Kconfig
index a9d159f50f..2d39d8e9f5 100644
--- a/sys/ztimer/Kconfig
+++ b/sys/ztimer/Kconfig
@@ -24,7 +24,7 @@ menu "Backends"
 config MODULE_ZTIMER_PERIPH_RTC
     bool "RTC peripheral"
     depends on HAS_PERIPH_RTC
-    select MODULE_PERIPH_RTC
+    select $(if-eq,$(BOARD),native,MODULE_ZTIMER_PERIPH_TIMER,MODULE_PERIPH_RTC)
 
 config MODULE_ZTIMER_PERIPH_RTT
     bool "RTT peripheral"
```

Reply:
I think #17165 should provide a better fix

Reply:
> I think #17165 should provide a better fix

Indeed, it works when I rebase this PR on top of #17165:

<details>

```
$ BUILD_IN_DOCKER=1 TEST_KCONFIG=1 BOARD=native make -C tests/periph_rtc clean all test --no-print-directory 
=== [ATTENTION] Testing Kconfig dependency modelling ===
=== [ATTENTION] Testing Kconfig dependency modelling ===
make: Nothing to be done for 'clean'.
Launching build container using image "riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/work/riot/RIOT:/data/riotbuild/riotbase:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'      -e 'BOARD=native'  -w '/data/riotbuild/riotbase/tests/periph_rtc/' 'riot/riotbuild:latest' make     all
/data/riotbuild/riotbase/makefiles/kconfig.mk:105: Warning! SHOULD_RUN_KCONFIG is not set but a previous configuration file was detected (did you run `make menuconfig`?). Kconfig will run regardless.
/data/riotbuild/riotbase/makefiles/kconfig.mk:105: Warning! SHOULD_RUN_KCONFIG is not set but a previous configuration file was detected (did you run `make menuconfig`?). Kconfig will run regardless.
Building application "tests_periph_rtc" for "native" with MCU "native".

"make" -C /data/riotbuild/riotbase/boards/native
"make" -C /data/riotbuild/riotbase/boards/native/drivers
"make" -C /data/riotbuild/riotbase/core
"make" -C /data/riotbuild/riotbase/cpu/native
"make" -C /data/riotbuild/riotbase/cpu/native/periph
"make" -C /data/riotbuild/riotbase/cpu/native/stdio_native
"make" -C /data/riotbuild/riotbase/drivers
"make" -C /data/riotbuild/riotbase/drivers/periph_common
"make" -C /data/riotbuild/riotbase/sys
"make" -C /data/riotbuild/riotbase/sys/auto_init
"make" -C /data/riotbuild/riotbase/sys/div
"make" -C /data/riotbuild/riotbase/sys/frac
"make" -C /data/riotbuild/riotbase/sys/test_utils/interactive_sync
"make" -C /data/riotbuild/riotbase/sys/xtimer
"make" -C /data/riotbuild/riotbase/sys/ztimer
   text	   data	    bss	    dec	    hex	filename
  43295	    736	  47976	  92007	  16767	/data/riotbuild/riotbase/tests/periph_rtc/bin/native/tests_periph_rtc.elf
r
=== [ATTENTION] Testing Kconfig dependency modelling ===
/work/riot/RIOT/tests/periph_rtc/bin/native/tests_periph_rtc.elf /dev/ttyACM0 
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

Help: Press s to start test, r to print it is ready
READY
s
START
main(): This is RIOT! (Version: 2022.01-devel-429-g1f40c-pr/cpu/native_rtc_ztimer_fix)

RIOT RTC low-level driver test
This test will display 'Alarm!' every 2 seconds for 4 times
Native RTC initialized.
  Setting clock to   2020-02-28 23:59:57
Clock value is now   2020-02-28 23:59:57
  Setting alarm to   2020-02-28 23:59:59
   Alarm is set to   2020-02-28 23:59:59
  Alarm cleared at   2020-02-28 23:59:57
       No alarm at   2020-02-28 23:59:59
  Setting alarm to   2020-02-28 23:59:61

Alarm!
Alarm!
Alarm!
Alarm!

```

</details>